blockchain/v0/reactor.go

```diff
@@ -368,7 +368,18 @@ FOR_LOOP:
 			firstParts := first.MakePartSet(types.BlockPartSizeBytes)
 			firstPartSetHeader := firstParts.Header()
 			firstID := types.BlockID{Hash: first.Hash(), PartSetHeader: firstPartSetHeader}
-			firstStateID := types.StateID{LastAppHash: first.Header.AppHash}
+
+			firstStateID := types.StateID{Height: first.Height - 1, LastAppHash: first.AppHash}
+			if first.Height == bcR.initialState.InitialHeight {
+				// Genesis state ID should have height 0
+				firstStateID.Height = 0
```

Comment:
Wouldn't it have genesis height, not height 0? (most often they are the same, but sometimes in the event of a hard fork because of a failure they are not.

Reply:
I tried to apply this feedback, but I see that using height 0 for "initial commit" is used quite often in the code.

For example, commit for first block should contain height 0, not initial height: https://github.com/dashevo/tenderdash/blob/adcc5817f13ff706fd74d52df903870c047c2a59/types/block.go#L796